src/factory/fetcher-factory.js

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
I don't disagree, but I wanted it to match up with the structure of `options`:

```
  let defaultOptions = {
    crawler: {
      ...
    },
    fetcher: {
      ...
    },
    ...
  }
```

We should consider changing "fetcher" to "githubFetcher" everywhere.